bot/exts/utilities/color.py

```diff
@@ -0,0 +1,211 @@
+import colorsys
+import json
+import logging
+import re
+from io import BytesIO
+
+from PIL import Image, ImageColor
+from discord import Embed, File
+from discord.ext import commands
+from rapidfuzz import process
+
+from bot.bot import Bot
+from bot.constants import Colours
+
+
+logger = logging.getLogger(__name__)
+
+
+ERROR_MSG = """The color code {user_color} is not a possible color combination.
+\nThe range of possible values are:
+\nRGB & HSV: 0-255
+\nCMYK: 0-100%
+\nHSL: 0-360 degrees
+\nHex: #000000-#FFFFFF
+"""
+
+COLOR_LIST = "bot/resources/utilities/ryanzec_colours.json"
+with open(COLOR_LIST) as f:
+    COLOR_JSON = json.load(f)
+
+
+# define color command
+class Color(commands.Cog):
+    """User initiated command to receive color information."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @commands.command(aliases=["colour"])
+    async def color(self, ctx: commands.Context, mode: str, *, user_color: str) -> None:
+        """Send information on input color code or color name."""
+        logger.info(f"{mode = }")
+        logger.info(f"{user_color = }")
+        if mode.lower() == "hex":
+            hex_match = re.search(r"^#(?:[0-9a-fA-F]{3}){1,2}$", user_color)
+            if hex_match:
+                hex_color = int(hex(int(user_color.replace("#", ""), 16)), 0)
```

Comment:
You're doing the same work twice here. You can use the regex match (which I'd suggest doing with `re.fullmatch` rather than `re.search` here. Also eliminates the need for the `^` and `$`) with grouping, and then pull the values for the conversion from the group. This will allow you to easily make the `#` optional.

Reply:
I went ahead and changed to re.fullmatch, but the second line is to convert the user input string hex color to a format that Discord can understand it. For some reason the embed expects a single integer value for the hex color or RGB color.